GVFS/GVFS.Common/LibGit2RepoPool.cs

```diff
@@ -1,47 +1,66 @@
 ﻿using GVFS.Common.Git;
 using GVFS.Common.Tracing;
 using System;
-using System.Collections.Concurrent;
 using System.Threading;
 
 namespace GVFS.Common
 {
     public class LibGit2RepoPool

```

Comment:
If we are going with a single object can we get rid of this class and have the single object and this code live in the `GitRepo`?  I would at least rename this class to `LibGit2RepoInvoker` or something similar since it isn't a pool anymore.